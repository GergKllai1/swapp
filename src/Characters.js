import React, { Component } from 'react'
import { View, ScrollView, StyleSheet } from 'react-native'
import axios from 'axios'
import CharacterCard from './CharacterCard';

const URL = 'https://swapi.co/api/'

export class Characters extends Component {
    state = {
        characters: [{name: "Anders"}]
    }

    async componentDidMount() {
        const response = await axios.get(URL + 'people')
        const characters = response.data.results;
        this.setState({
            characters: characters
        })
    }
  render() {
    const CharacterDisplay = this.state.characters.map(character => {
        return <CharacterCard testID={character.name} key={character.name} navigation = {this.props.navigation} character = {character} />
    })
    return (
      <ScrollView testID='main' style={styles.bigBlue}>
        <View style={styles.position}>
        {CharacterDisplay}
        </View>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  bigBlue: {
    color: 'blue',
    fontWeight: 'bold',
    fontSize: 30,
    textAlign: 'center'
  },
  position: {
    flex: 1, 
    justifyContent: 'center', 
    alignItems: 'center',
    marginTop: 30
  },
});

export default Characters
