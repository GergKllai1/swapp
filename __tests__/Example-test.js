import React from 'react';
import { shallow, configure } from 'enzyme';
import Characters from '../src/Characters';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe('Testing ReassignLocationMenu component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Characters />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});