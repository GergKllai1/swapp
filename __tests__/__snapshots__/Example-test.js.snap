// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReassignLocationMenu component renders as expected 1`] = `
<ScrollViewMock
  style={
    Object {
      "color": "blue",
      "fontSize": 30,
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <CharacterCard
      character={
        Object {
          "name": "Anders",
        }
      }
      key="Anders"
    />
  </Component>
</ScrollViewMock>
`;
